[Quek Jing Ling, Brian] iP

astraxq opened the pull request:
# PepeBot
> Your very first to-do bot that keeps track of all your tasks effortlessly! - ~~Somebody~~ Pepe :frog:

## Features
- Able to keep track of your tasks with Pepe's task tracker!
- Has a wide variety type of tasks to cover all sorts of tasks!

## Download Guide:
1. Click on this [link ](https://github.com/nus-cs2103-AY2223S2/ip/pull/251)
2. Add your tasks
3. Mange it!

And it is **FREE**!

## Task List:
- [x] toDo
- [x] Deadlines
- [x] Events
- [x] Mark/UnMark
- [x] GUI
---
If you are a Java Programmer, you can reproduce more bot commands too! Heres the ```getResponse(String input)``` method:

``` Java
   public String getResponse(String input) {
        String output = "";
        try {
            output += Parser.readCommand(input, tasks);
            Storage.writeData(tasks);
        } catch (DukeException e) {
            output += e.toString();
        }
        return output;
    }
```

Commit:
Add Gradle support

astraxq committed:
feat: greet, echo, exit

astraxq committed:
Revert "feat: greet, echo, exit"

This reverts commit 4a7d4b66726cd32f839df8f9a4aa47468a6b9ed7.

astraxq committed:
feat: greet, echo, exit

astraxq committed:
fix: redundant code

astraxq committed:
feat: add, list

Commit:
fix: index bug when calling list

Commit:
feat: mark as done

Commit:
Level-4
feat: todo, deadline, event obj implemented

Commit:
feat: add todo command into bot

Commit:
add exception handling

astraxq committed:
Level-5
feat: refined exception handling

astraxq committed:
Level-6
feat: delete functionality

astraxq committed:
add JavaDocs and Packages

Commit:
add fileparser and filecreate classes

astraxq committed:
refactoring codebase

astraxq committed:
add save feature with data storing capabilities

astraxq committed:
refactor packages name

astraxq committed:
Revert "Level-6"

This reverts commit dc8a6c85ee2681c577bee93d7053755bd1b2348f.

astraxq committed:
refactoring Duke.java

astraxq committed:
merge branch-7 save feature into master

astraxq committed:
fix Duke.java bugs from Level-6

astraxq committed:
add date field in deadline object

astraxq committed:
merge branch-8 date and time feature into master

astraxq committed:
fix deadline and date so textparser can read

astraxq committed:
add Storage class

astraxq committed:
add serializable interface in task objects

astraxq committed:
fix storage to read object inputs and outputs

astraxq committed:
remove serializable in sub task types

astraxq committed:
add parser class

astraxq committed:
add deadline and event logic into parser

astraxq committed:
add UI class

astraxq committed:
merge add-gradle-support branch into master

astraxq committed:
fix packages location

astraxq committed:
fix storage path

astraxq committed:
add JUnit test methods

astraxq committed:
add JAR in gradle

HmuuMyatMoe commented on src/main/java/duke/duke_exception/DukeException.java:
```diff
@@ -0,0 +1,66 @@
+    /**
+     * Constructor for the DukeException object.
+     */
+    public DukeException(int index) {
```
How did you use / do you plan to use this particular constructor for DukeException in the code?

astraxq replied:
Apologies I've removed this constructor and only plan to use the constructor that takes in the message string

```
    public DukeException(String msg) {
        this.message = msg;
    }
```

HmuuMyatMoe commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,69 @@
+
+public class TaskList implements Serializable {
+    private static DukeException invalidIndex = new DukeException("Invalid index keyed.");
+    private ArrayList<Task> list = new ArrayList<>();
```
Suggestion to use plural form for names of a collection of objects.
eg. int[] values;

astraxq replied:
Good catch! Thank you so much.

HmuuMyatMoe commented on src/main/java/duke/utility/ui/Ui.java:
```diff
@@ -0,0 +1,48 @@
+            + "|____/ \\__,_|_|\\_\\___|\n";
+    private static String openingMessage = "Welcome to the Duke Bot.\n";
+
+    public static void welcomeMessage() {
```
Great job on using verbs and camelCase for method names!

HmuuMyatMoe commented on src/main/java/duke/tasklist/task_types/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+        return String.format("[D]%s (by: %s)", super.toString(), this.deadline.dateToString());
+    }
+
+    public String toTextString() {
```
Perhaps a more intuitive name that suggests that the method is formatting the string to be stored in the data file would make code easier to understand.

astraxq replied:
I agree. `formatStringToData` is definitely more intuitive.

HmuuMyatMoe commented on src/main/java/duke/duke_exception/DukeException.java:
```diff
@@ -0,0 +1,66 @@
+        }
+    }
+
+    public DukeException(String msg) {
```
Add java doc comments for all public classes and methods.

astraxq replied:
I have done so in subsequent commits!

HmuuMyatMoe reviewed (commented):
Code is generally quite neat. Only slight modifications required.
Great job on the variables and method names.

Commit:
remove toTextString method as we use object stream

Commit:
change the storage file location

Commit:
add JavaDocs

Commit:
merge JavaDoc branch into master

Commit:
fix coding standards based on google styling

Commit:
add coding standards based on google standards

Commit:
merge CodingStandard branch into master

Commit:
add search command into duke bot

Commit:
merge Level-9 branch into master

Commit:
fix naming convention and bugs mention by reviewer

joshua-tyh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,45 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Scanner inputScanner = new Scanner(System.in);
```
As the UI class serves to deal with interactions with the user, maybe the scanner that reads user inputs could be abstracted into the UI class?

Additionally, you could also consider abstracting print statements, like the logo printing at lines 5-8, into the UI class as well.

astraxq replied:
1. I feel that UI's main job should be dealing with what to printout and not the interaction as itself. I didn't want to put too many things into the UI class itself but I will take note and adjust accordingly at the very end of the iP.

2. Yeap the 5-8 are removed line so they have been moved into the UI class.

joshua-tyh commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,144 @@
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import duke.duke_exception.DukeException;
```
Consider leaving a blank line separator at line 8 to follow the Java Coding Standards outlined for package imports to separate "java" imports and "duke" imports.

astraxq replied:
Sounds good.

joshua-tyh commented on src/main/java/duke/tasklist/task_types/Event.java:
```diff
@@ -0,0 +1,31 @@
+/**
+ * Represents a <code>task</code> object that contains a string holding the <code>event</code> time
+ * from a specific length stated by the user.
+ * 
```
Consider removing the blank line at line 6 to keep a consistent format for the header Javadoc of each class.

joshua-tyh commented on src/main/java/duke/tasklist/task_types/Task.java:
```diff
@@ -0,0 +1,47 @@
+ * Represents a <code>task</code> object that contains a string holding the <code>name</code> of the
+ * task and the <code>status</code> of the task. Represents a <code>task</code> object that contains
+ * a string holding the <code>name</code> of the task and the <code>status</code> of the task.
+ * 
```
Consider removing the blank line at line 9 to keep a consistent format for the header Javadoc of each class.

joshua-tyh commented on src/main/java/duke/tasklist/task_types/ToDo.java:
```diff
@@ -0,0 +1,26 @@
+
+/**
+ * Represents a <code>task</code> object with the todo tag into it.
+ * 
```
Consider removing the blank line at line 5 to keep a consistent format for the header Javadoc of each class.

joshua-tyh commented on src/main/java/duke/utility/parser/CommandMap.java:
```diff
@@ -0,0 +1,12 @@
+package duke.utility.parser;
+
```
Consider removing the blank like at line 2 to ensure consistency across code files..

joshua-tyh commented on src/main/java/duke/utility/parser/Parser.java:
```diff
@@ -0,0 +1,256 @@
+import duke.tasklist.task_types.Task;
+import duke.tasklist.task_types.ToDo;
+import duke.utility.ui.UiMessage;
+
```
Consider removing the blank line at line 10 to ensure consistency across code files..

joshua-tyh commented on src/main/java/duke/utility/parser/Parser.java:
```diff
@@ -0,0 +1,256 @@
+    private static DukeException noSpecialParam =
+            new DukeException("Missing special param e.g /by, /from, /to");
+    private static DukeException emptyParam = new DukeException("Empty parameter inserted.");
+
```
Consider leaving only a single line separator between attributes and methods instead of 2 lines to ensure consistency across code files.

joshua-tyh reviewed (commented):
Overall, the code looks neat and in line with the Java Coding Standards. Just some slight adjustments could be required to ensure consistency across code files. Great work!

astraxq committed:
fix spacing for consistency

astraxq committed:
add javafx tutorial part 1

astraxq committed:
add JavaFX GUI into PepeBot

astraxq committed:
add javadoc for Duke class

astraxq committed:
add javadoc for affected classes that aids in FXML

astraxq committed:
merge Level-10 branch into master

astraxq committed:
Fix build.gradle for generating a new jar file

Implementing JavaFX has changed where the main func is.

The main function moved to Launcher class, mainClassName is then change,

astraxq committed:
fix: versioning of dialogbox

astraxq committed:
add assertion for long user inputs

astraxq committed:
Merge pull request #2 from astraxq/A-Assertions

Add assertions into User Input.

astraxq committed:
add code quality check on boolean status

"status" was not following the boolean-like naming convention.

It is too generic, hence changed to isMarked.

astraxq committed:
Merge pull request #3 from astraxq/A-CodeQuality

Add Code Quality Check

astraxq committed:
add CI/CD tools

astraxq committed:
Merge pull request #4 from astraxq/A-CI

add CI/CD tools

astraxq committed:
add sort command into command map

astraxq committed:
change feature to update

due to group member implementing sort, update is implemented.

astraxq committed:
refactor commandmap

astraxq committed:
add updatetype enums

astraxq committed:
add update methods in the respective task types

astraxq committed:
add update feature into bot

astraxq committed:
fix bot response such that it corresponds to list

astraxq committed:
Merge pull request #6 from astraxq/branch-Extension

Add update feature into bot

astraxq committed:
add javadocs

astraxq committed:
Merge pull request #7 from astraxq/branch-Extension

Added JavaDocs from Extension Features

astraxq committed:
fix: JavaDoc of TaskList class

astraxq committed:
feat: dynamic styling as window expands

astraxq committed:
fix: user image aligning

astraxq committed:
fix: scrollpane was not expanding properly

astraxq committed:
feat: add a help command to improve UX

astraxq committed:
feat: add new display pictures

astraxq committed:
Merge pull request #8 from astraxq/branch-BetterGui

Better GUI/UX

astraxq committed:
fix: bug where mark/unmark is not displayed

astraxq committed:
feat: add empty search result output

astraxq committed:
fix: delete output

where it tries to get the already deleted task.

astraxq committed:
fix: versioning of .fxml file

astraxq committed:
feat: add Ui.png

astraxq committed:
fix: .jar file name

astraxq committed:
fix: storage file location

astraxq committed:
feat: add exit upon saying bye

astraxq committed:
fix: update readme

astraxq committed:
fix: bug on empty parameter for mark/unmark

astraxq committed:
fix: update command syntax to be clearer

astraxq committed:
fix: update Readme with user guide &quick start

astraxq committed:
fix: ui.png pathing specification

astraxq committed:
fix: minor styling on readme

astraxq committed:
feat: add a table reference in readme
